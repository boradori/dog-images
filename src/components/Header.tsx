import React, { FC } from 'react'
import styled from '@emotion/styled'
import Icon from './Icon'

const Header: FC = () => {
  return (
    <Container>
      <Title>Dog Breeds</Title>
      <Icon icon="redHeartIcon" alt="red heart icon" />
    </Container>
  )
}

const Container = styled.div({
  display: 'flex',
  justifyContent: 'space-between',
  paddingLeft: '1rem',
  paddingRight: '1rem',
  maxWidth: '640px',
  width: '100%',
})

const Title = styled.h1({
  fontWeight: 'bold',
  fontSize: '1.5rem',
  lineHeight: '33px',
})

export default Header
