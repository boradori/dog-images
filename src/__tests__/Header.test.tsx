import React from 'react'
import { render, cleanup } from '@testing-library/react'
import Header from '../components/Header'

describe('<Header />', () => {
  afterEach(cleanup)

  it('renders and takes a snapshot', () => {
    const { asFragment } = render(<Header />)
    expect(asFragment).toMatchSnapshot()
  })

  it('renders title', () => {
    const { getByText } = render(<Header />)
    expect(getByText('Dog Breeds')).toBeInTheDocument()
  })

  it('renders icon', () => {
    const { getByAltText } = render(<Header />)
    expect(getByAltText('red heart icon')).toBeInTheDocument()
  })
})
