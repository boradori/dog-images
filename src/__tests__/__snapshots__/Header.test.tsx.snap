// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders 1`] = `[Function]`;

exports[`<Header /> renders and takes a snapshot 1`] = `[Function]`;
